File: Bit Manipulation/1318. Minimum Flips to Make a OR b Equal to c/solution.ts
function minFlips(a, b, c) {
    let flips = 0;
    for (let i = 0; i < 31; i++) {
        let bitA = a & 1, bitB = b & 1, bitC = c & 1;
        if (bitC === 0) {
            flips += bitA + bitB;
        } else {
            flips += bitA === bitB && bitB === 0 ? 1 : 0;
        }
        a >>= 1; b >>= 1; c >>= 1;
    }
    return flips;
}

File: Bit Manipulation/1318. Minimum Flips to Make a OR b Equal to c/solution.py
class Solution:
    def minFlips(self, a, b, c):
        flips = 0
        for i in range(31):
            bitA, bitB, bitC = a & 1, b & 1, c & 1
            if bitC == 0:
                flips += bitA + bitB
            else:
                flips += 1 if bitA == bitB == 0 else 0
            a >>= 1; b >>= 1; c >>= 1
        return flips
